### PR TITLE
Jetpack Cloud: Add LogRocket

### PR DESCRIPTION
**Pull request by Initsogar**
Proof of concept PR for Log Rocket. More details in p1HpG7-rVk-p2.

## Proposed Changes

* Add and register LogRocket as a tracking tool
* Add a `logrocket` feature flag and enable it on Jetpack Cloud Production.
* Add LogRocket user identification for logged-in users, using hashed PII for privacy.
* Initialize LogRocket only on Backup, Server Credentials settings and Activity Log pages.
* Disable all inputs from recording to ensure sensitive data is not being sent to LogRocket. Here is an example of how input fields will look:
  

## Testing Instructions
* Spin up a Jetpack Cloud Live branch and append `?flags=ad-tracking,logrocket` to the URL to enable the feature flags.
* Enable debug logs using the developer console:
  `localStorage.setItem( 'debug', 'calypso:*' );`
* Filter console and network tabs by `logrocket`
* Navigate to the `Backup` or `Settings` page and ensure you see these debug records:
   CleanShot 2024-04-08 at 00 04 11@2x
* Play around checking your backups, downloading some files, updating your server credentials or even change your retention period.
* And check the network tab and there should be some calls to LogRocket.
* After some time, the session recorded should be available in LogRocket dashboard. If you want access to the platform to watch the recording, please ask for access in p1HpG7-rVk-p2.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/advanced-credentials/credentials-form/index.tsx [hunk 1]
```diff
@@ -217,6 +217,7 @@ const ServerCredentialsForm: FunctionComponent< Props > = ( {
 					isError={
 						formErrors.kpri && ( interactions.kpri || ! formErrors.kpri.waitForInteraction )
 					}
+					data-private
```
Comment (Initsogar) [inline client/components/advanced-credentials/credentials-form/index.tsx]: This is necessary to prevent LogRocket record this private key field

Comment (Initsogar) [reply client/components/advanced-credentials/credentials-form/index.tsx]: CleanShot 2024-04-07 at 20 57 57@2x

Comment (Initsogar) [reply client/components/advanced-credentials/credentials-form/index.tsx]: No longer needed, we are marking all input elements as private by default.

File: client/components/advanced-credentials/index.tsx [hunk 2]
```diff
@@ -412,6 +412,10 @@ const AdvancedCredentials: FunctionComponent< Props > = ( {
 		}
 	};
 
+	useEffect( () => {
+		dispatch( loadTrackingTool( 'LogRocket' ) );
+	}, [ dispatch ] );
+
```
Comment (elliottprogrammer) [inline client/components/advanced-credentials/index.tsx:418]: Nitpick: I wasn't expecting to see/find a `useEffect` way down here, alone by itself, just before the component render/return. I mean, it's fine I guess, functionality and logic works just fine, I'm just used to seeing useEffect functions usually grouped somewhat together, higher up in the component code. Just thought I'd mention it. 🙂

**Review by elliottprogrammer — APPROVED** [review]
Tested and works good as described (although I had to force myself into a non-STS (session tracking sensitive) zone). 
The code looks good to me overall. I don't see any issues, just a couple very minor nitpicks that I mentioned.
LGTM! 👍

P.S. - Apologies for the delay reviewing, we were at a meetup. 🙂

File: client/components/jetpack/daily-backup-status/status-card/backup-just-completed.tsx [hunk 3]
```diff
@@ -23,6 +25,10 @@ const BackupJustCompleted: React.FC< Props > = ( { justCompletedBackupDate, last
 	const justCompletedDisplayDate = getDisplayDate( justCompletedBackupDate, false );
 	const lastBackupDisplayDate = getDisplayDate( lastBackupDate, false );
 
+	useEffect( () => {
+		recordLogRocketEvent( 'calypso_jetpack_backup_just_completed' );
+	} );
```
Comment (elliottprogrammer) [inline client/components/jetpack/daily-backup-status/status-card/backup-just-completed.tsx]: Same nitpick here (about including the empty dependency array) - See below.

File: client/components/jetpack/daily-backup-status/status-card/backup-in-progress.tsx [hunk 4]
```diff
@@ -26,6 +28,10 @@ const BackupInProgress: React.FC< Props > = ( { percent, inProgressDate, lastBac
 		? getDisplayDate( lastBackupDate, false )
 		: undefined;
 
+	useEffect( () => {
+		recordLogRocketEvent( 'calypso_jetpack_backup_in_progress' );
+	} );
```
Comment (elliottprogrammer) [inline client/components/jetpack/daily-backup-status/status-card/backup-in-progress.tsx]: Another nitpick:  I know not including the useEffect dependency array (useEffect second argument) defaults to an empty array, still I think it's more explicit, clear, and readable to include it. Just my opinion. Change or not change at your discretion. 🙂

File: client/components/advanced-credentials/index.tsx (hunk 2 above)
Comment (Initsogar) [reply client/components/advanced-credentials/index.tsx:418]: I understand what you are saying and makes totally sense. I preferred to isolate it from other parts of that component for this specific experiment. I think we could decide better places to load the tracking tool once we finish with the trial.

File: client/components/jetpack/daily-backup-status/status-card/backup-in-progress.tsx (hunk 4 above)
Comment (Initsogar) [reply client/components/jetpack/daily-backup-status/status-card/backup-in-progress.tsx]: You are right! It is better to be explicit. I just added the empty dependency array for both cases. Thanks a lot!

**Review by elliottprogrammer — APPROVED** [review]

**Comment by Initsogar** [comment]
> Tested and works good as described (although I had to force myself into a non-STS (session tracking sensitive) zone). The code looks good to me overall. I don't see any issues, just a couple very minor nitpicks that I mentioned. LGTM! 👍
> 
> P.S. - Apologies for the delay reviewing, we were at a meetup. 🙂

@elliottprogrammer Thank you so much!!! I really appreciate your support on this one!
I just pushed the changes for some nitpicks and I will proceed with the merge.

**Pull request merged**